src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -292,5 +306,72 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
             result = results;
             return true;
         }
+
+        internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            if (!TryReadFile(statusFilePath, reusableReader, out string statusFileContents))
+            {
+                // Between the time that we get an ID and the time that we try to read the associated stat
+                // file(s), the process could be gone.
+                result = default(ParsedStatus);
+                return false;
+            }
+
+            var parser = new StringParser(statusFileContents, '\n');
+            var results = default(ParsedStatus);
+
+            while (parser.MoveNext())
+            {
+                string[] elements = parser.ExtractCurrent().Split(Delimiters, 3, StringSplitOptions.RemoveEmptyEntries);
+                if (elements.Length < 2)
+                {
+                    continue;
+                }
+
+                switch (elements[0])
+                {
+                    case "Pid":
+                        results.Pid = int.Parse(elements[1]);
```

Comment:
We should use TryParse rather than Parse, for all of these Parse calls.  If there's an issue with the incoming content for whatever reason, we don't want that to throw an exception.

Reply:
Done.